packages/compass/src/app/intercom/setup-intercom.ts

```diff
@@ -0,0 +1,88 @@
+import type { EventEmitter } from 'events';
+import createLoggerAndTelemetry from '@mongodb-js/compass-logging';
+import type { IntercomMetadata } from './intercom-script';
+import { IntercomScript, buildIntercomScriptUrl } from './intercom-script';
+
+const { debug } = createLoggerAndTelemetry('COMPASS-INTERCOM');
+
+type User = {
+  id: string;
+  createdAt: Date;
+};
+
+type Preferences = Pick<EventEmitter, 'on'> & {
+  isFeatureEnabled: (feature: string & 'enableFeedbackPanel') => boolean;
```

Comment:
This is just a type cleanup, because `string & 'enableFeedbackPanel'` is already the same type as `'enableFeedbackPanel'`.

But that’s okay, because `isFeatureEnabled: (feature: string) => boolean` is assignable to `(feature: 'enableFeedbackPanel') => boolean` – function arguments are contravariant, i.e. making an argument type more specific is actually making the requirements on the function itself *less* specific.

Reply:
Nice TIL, i thought that it would have only accepted `'enableFeedbackPanel'` as the type for that argument then, thanks for the explanation!